zebra-scan/src/service.rs

```diff
@@ -75,6 +72,10 @@ impl Service<Request> for ScanService {
     }
 
     fn call(&mut self, req: Request) -> Self::Future {
+        if let Err(error) = req.check() {
+            return async move { Err(error) }.boxed();
+        }
```

Comment:
Optional: Can we create a test that triggers this error by sending more than 1000 keys or 0 to some of the service calls ? 

Don't need to be done in this PR, we can open a ticket.

Reply:
Yep, added in https://github.com/ZcashFoundation/zebra/pull/8219/commits/70e176e748f415cb6d977b7e09c93fd802a5b6f3.